### PR TITLE
[v6r20] [ALERT] better handling of return values (API and TaskManager)

### PR DESCRIPTION
**Pull request by fstagni**
@andresailer you better check this one

BEGINRELEASENOTES

*Interfaces
CHANGE: setParameterSequence always return S_OK/S_ERROR

*TS
FIX: check for return value on Job interface and handle it

ENDRELEASENOTES

### CODE REVIEW

**Comment by andresailer** [comment]
* The `return S_ERROR` in `setParameterSequence` should be `return _reportError` instead. (That appends errors to the Job.errorDict member, which is checked in Interfaces.API.Dirac.submitJob, then errors cannot be ignored because submission will not be done)
* There is a failing test now

File: TransformationSystem/Client/TaskManager.py [hunk 1]
```diff
@@ -798,7 +803,10 @@ def _handleRest(self, oJob, paramsDict):
         if paramValue:
           self._logDebug('Setting %s to %s' % (paramName, paramValue),
                          transID=transID, method='handleRest')
-          oJob._addJDLParameter(paramName, paramValue)
+          res = oJob._addJDLParameter(paramName, paramValue)
```
Comment (andresailer) [inline TransformationSystem/Client/TaskManager.py]: This makes at least one of the tests fail. `_addJDLParameter` returns `None`

**Comment by fstagni** [comment]
I updated according to review. Anyway my [ALERT] is because the `setParameterSequence` will now always return `S_OK/S_ERROR`, while before it was returning either `None` or `S_ERROR`. This corrects a bug but may introduce some issues on your side if you were checking the output of it.

**Comment by fstagni** [comment]
I am not entirely sure why the other test is failing (if it's "genuine" or not). I will look at it on Monday.

**Comment by andresailer** [comment]
we are not checking the return value from `setParameterSequence`,

**Pull request merged**